### DISCUSSION TITLE
Idiomatic way to get enter/exist events for widgets.

### DISCUSSION THREAD

**Discussion by madelaney** [emilk/egui#1712] [Q&A] [answered]
I was wondering what is the best way to get events (for example mouse enter / mouse exist) on widgets within an app.

**Comment by willser** [comment, marked as answer]
Hi,you should check out this [site](https://www.egui.rs/#demo).

There is a window which name is `Inspection`.It will let you find out [these code](https://github.com/emilk/egui/blob/935913b1ec5fb95c468fde60a107daf4ac164a9e/egui/src/context.rs#L1108) in this repository
